regexp-tests/openconfig-inet-types-test.yang

```diff
@@ -16,17 +16,45 @@ module openconfig-inet-types-test {
   }
   leaf ipv4-address-zoned {
     type ocinet:ipv4-address-zoned;
+    // Invalid chars and base tests
     pt:pattern-test-pass "255.1.1.1%eth0";
-    pt:pattern-test-pass "255.255.255.255%eth1";
     pt:pattern-test-pass "0.0.0.0%PoRt3";
     pt:pattern-test-pass "1.1.1.1%FOX10_mouse5";
-    pt:pattern-test-fail "255.1.1.1%";
-    pt:pattern-test-fail "255.255.255.255";
-    pt:pattern-test-fail "0.0.0.0";
-    pt:pattern-test-fail "1.1.1.1";
-    pt:pattern-test-fail "256.255.255.255";
-    pt:pattern-test-fail "1.1.1.256";
-    pt:pattern-test-fail "256.1.1.1%eth0";
+    pt:pattern-test-fail "10.1.1.1";
+    pt:pattern-test-fail "10.1.1.1%";
+    pt:pattern-test-fail "10.1.1.1%zone#1";
+    pt:pattern-test-fail "10.1.1.1%eth0@";
+    pt:pattern-test-fail "10.1.1.1%eth0!";
+    pt:pattern-test-fail "10.1.1.1%eth0$";
+    pt:pattern-test-fail "10.1.1.1%eth0 ";
+    // ADDRESS%(IF_NAME)?[:](SUB_IF_NAME) tests
+    pt:pattern-test-pass "10.10.10.10%ns:eth0";
+    pt:pattern-test-pass "10.10.10.10%abc:123";
+    pt:pattern-test-fail "10.10.10.10%:";
+    pt:pattern-test-fail "10.10.10.10%:eth0";
+    pt:pattern-test-fail "10.10.10.10%eth0:";  
+    pt:pattern-test-fail "10.10.10.10%:.-";
+    // ADDRESS%(IF_NAME)?[.](SUB_IF_NAME) tests
+    pt:pattern-test-pass "172.16.0.1%Ethernet1.1";
+    pt:pattern-test-pass "10.0.0.1%abc.123";
+    pt:pattern-test-fail "10.0.0.1%.";
+    pt:pattern-test-fail "10.0.0.1%eth0.";
+    pt:pattern-test-fail "10.0.0.1%.eth0";
+    pt:pattern-test-fail "10.0.0.1%..";
+    // ADDRESS%(IF_NAME)?[/](SUB_IF_NAME) tests
+    pt:pattern-test-pass "192.168.1.1%Ethernet1/2";
+    pt:pattern-test-pass "10.10.10.10%uplink/4";
+    pt:pattern-test-fail "10.10.10.10%/";
+    pt:pattern-test-fail "10.10.10.10%/uplink";
+    pt:pattern-test-fail "10.10.10.10%uplink/";
+    pt:pattern-test-fail "10.10.10.10%uplink//2";
+    // ADDRESS%(IF_NAME)?[-](SUB_IF_NAME) tests
```

Comment:
This should also add a set of test strings for `_` since that is allowed in the pattern